### PR TITLE
Added Edit Distance Algorithm in String Algorithms

### LINKED ISSUE #97: Edit distance algorithm
I would like to implement edit distance algorithm for strings

### PR DESCRIPTION
**Pull request by mayanktaluja**
Closes #97.

### CODE REVIEW

File: C++/source/string/edit_distance.cpp [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+Time Complexity :- O(m * n)
+Space Complexity :- O(m * n)
+                    Where m = length of string1 and n = string2
+
+*/
+
+
+
+#include<bits/stdc++.h>
```
Comment (faheel) [inline C++/source/string/edit_distance.cpp:15]: It's good practice to only `#include` those headers that are required.

Also, this header is not present on OS X, as so the code does not compile on it.

File: C++/source/string/edit_distance.cpp [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+using namespace std;
+ 
+int min(int x, int y, int z) {
+   return min(min(x, y), z);
+}
```
Comment (faheel) [inline C++/source/string/edit_distance.cpp:21]: No need for this function. You can instead use `min({some_val, some_other_val, yet_another_val})` in your code.

File: C++/source/string/edit_distance.cpp [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    return 1 + min ( editDist(str1,  str2, m, n-1),    
+                     editDist(str1,  str2, m-1, n),   
+                     editDist(str1,  str2, m-1, n-1)
+                   );
+}
+ 
+int main()
+{
+    string str1 = "sunday";
+    string str2 = "saturday";
```
Comment (faheel) [inline C++/source/string/edit_distance.cpp:40]: To make it more dynamic, please take the input for `str1` and `str2` from the user.

File: C++/source/string/edit_distance.cpp [hunk 4]
```diff
@@ -0,0 +1,45 @@
+/*
+
+Edit Distance Algorithm :-
+--------------------------------
+Description: Find the minimum number of operations required to convert one string into another.
+
+Time Complexity :- O(m * n)
+Space Complexity :- O(m * n)
+                    Where m = length of string1 and n = string2
```
Comment (faheel) [inline C++/source/string/edit_distance.cpp:9]: The space complexity is O(1).

**Review by faheel — CHANGES_REQUESTED** [review]

**Comment by alxmjo** [comment]
Looks like this hasn't been updated in a few months. Closing.

**Pull request closed without merging**